### PR TITLE
Added Parameters to routing

### PR DESCRIPTION
**Pull request by Meggie-H**
I also added QueryClientProvider settings by accident. I'm sorry

### CODE REVIEW

File: src/routes/$username.Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { createFileRoute } from '@tanstack/react-router';
+import Dashboard from '../components/Dashboard';
+
+export const Route = createFileRoute('/$username/Dashboard')({
+  component: () => <Dashboard/>,
```
Comment (yanevdw) [inline src/routes/$username.Dashboard.tsx]: You could also do `component: Dashboard` here.

File: src/components/GitGraph.tsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import { Gitgraph } from "@gitgraph/react";
 import { getCommits } from "../services/apiService";
 import { ICommit } from "../models";
-import { useMutation, useQuery } from "react-query";
-import { useQueryClient } from "@tanstack/react-query";
+import { useMutation  } from "react-query";
```
Comment (yanevdw) [inline src/components/GitGraph.tsx]: I see you have imported this, but it's never used in the file.

**Review by yanevdw — APPROVED** [review]
The routing has been implemented really well. Well done!

File: src/components/GitGraph.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 import ReadMore from "./ReadMore";
+import { useParams } from "@tanstack/react-router";
 
 // create an animation for the git graph
 function GitGraph() {
+  const { username } = useParams({ strict: false });
+
   const { data: commitData, isLoading, isError } = useQuery({
     queryKey: [`getCommits`],
     queryFn: () => getCommits('Hugovs2000', 'React-Project'),
```
Comment (theadley) [inline src/components/GitGraph.tsx:14]: Assuming this won't be hardcoded in the future?

Comment (theadley) [reply src/components/GitGraph.tsx:14]: As you now have a username param, so maybe using that to get the repos of the username and then get the commits of the selected username?

**Review by theadley — APPROVED** [review]
Looks good, I just had questions around hardcoding which repo you fetch

File: src/components/GitGraph.tsx (hunk 3 above)
Comment (Meggie-H) [reply src/components/GitGraph.tsx:14]: Yes, so I'm still working on the component where the user will be able to pick the repo. I'll add that repo as another param and then use them here to get the repo data.

**Pull request merged**